src/core/filesystem.cc

```diff
@@ -1175,6 +1180,40 @@ S3FileSystem::ParsePath(
   return Status::Success;
 }
 
+Status
+S3FileSystem::CleanPath(const std::string& path, std::string* clean_path)
+{
+  // Remove extra slashes
+  *clean_path = "s3://";
+  int true_length = path.size();
+  for (size_t i = path.size() - 1; i++) {
+    if (path[i] == '/') {
+      true_length -= 1;
+    }
+  }
+  std::string true_path = path.substr(0, true_length);
```

Comment:
This seems unnecessarily complex, can you just use something like this: https://www.techiedelight.com/trim-string-cpp-remove-leading-trailing-spaces/ except for '/' instead of whitespace

Reply:
The function needs to convert **abc.com:9000/testbucket//object//aa/** ->**abc.com:9000/testbucket/object/aa**